### DISCUSSION TITLE
Npm install is taking 10+ minutes to finish

### DISCUSSION THREAD

**Discussion by BioCarmen** [community/community#26648] [Actions]
<p>Hey there.<br>
I have been using github actions for all my tests.<br>
<code>npm install</code> step only took 1 minute last week, but these couple days it took 10+ minutes to just install. Any one experience the same issue?</p>

**Comment by guido-pti** [comment]
<p>same here, npm install or ci takes &gt; 5 min, where it took normally 30 sec.<br>
Any idea on the issue</p>

**Comment by timbeks** [comment, marked as answer]
<p>After some digging around, it appears NPM uses the old and discontinued <code>git://</code> protocol. See <a href="https://github.blog/2021-09-01-improving-git-protocol-security-github/" rel="noopener nofollow ugc">https://github.blog/2021-09-01-improving-git-protocol-security-github/</a>.</p>
<p>The fix we are using is:</p>
<pre><code class="lang-auto">git config --global url."git@github.com:".insteadOf git://github.com/
</code></pre>
<p>This will tell git to use ssh instead of the old <code>git://</code> protocol. Hope this can help others.<br>
If there is an even better solution, I would love to hear about it!</p>

**Reply by datatrigger** [reply]
Did not work for me on GitHub Codespaces. See my comment below.

**Comment by BioCarmen** [comment]
<p>Thanks <a class="mention" href="/timbeks">@timbeks</a> How do you set it? It there a global way to set it? I don’t want to add it at each step of my workflows</p>

**Comment by timbeks** [comment]
<p>Hi <a class="mention" href="/biocarmen">@BioCarmen</a> ,</p>
<p>We’ve put that command within our base images. I don’t know how to resolve this with Github Actions.</p>
<p>Hopefully you can do the same.</p>

**Comment by mike-kaminski** [comment]
<p>The most direct approach would be stick it as a step just before your npm install step. Or as part of the NPM install step like so: <code>git config --global url."git@github.com:".insteadOf git://github.com/ &amp;&amp; npm install</code>. No need for adding it to every step, just the npm install.</p>
<p>You could alternatively set up a customized build container with any specialized configuration or tools pre-installed and then <a href="https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container" rel="noopener nofollow ugc">run the job in that</a>. I’ve found this method very helpful in smoothing out and speeding up workflows that are dependency heavy (package deps, custom config, custom security <em>stuff</em>).</p>

**Comment by rtabulov** [comment]
is this problem still relevant today?

**Comment by billyjov** [comment]
The problem is still relevant. I'm facing similar issue since a couple of days.

**Comment by datatrigger** [comment]
Hello, I had the same problem in GitHub Codespaces. 7+ minutes to `npm install` cors, Express.js and Socket.IO. Now it went down to ~10 seconds with the following command: `npm install --registry=https://registry.npmmirror.com`
